tasks/function_test_groups.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import execa from 'execa';
+import grunt from 'grunt';
+
+/**
+ * The list of tags that we use in the functional tests, if we add a new group we need to add it to this list
+ * and to the list of jobs in .ci/jobs.yml
+ */
+const TEST_TAGS = [
+  'ciGroup1',
+  'ciGroup2',
+  'ciGroup3',
+  'ciGroup4',
+  'ciGroup5',
+  'ciGroup6',
+  'ciGroup7',
+  'ciGroup8',
+  'ciGroup9',
+  'ciGroup10',
+  'ciGroup11',
+  'ciGroup12'
+];
+
+export function getFunctionalTestGroupRunConfigs({ esFrom, kibanaInstallDir } = {}) {
+  return {
+    // include a run task for each test group
+    ...TEST_TAGS.reduce((acc, tag) => ({
+      ...acc,
+      [`functionalTests_${tag}`]: {
+        cmd: process.execPath,
+        args: [
+          'scripts/functional_tests',
+          '--include-tag', tag,
+          '--config', 'test/functional/config.js',
+          '--esFrom', esFrom,
+          '--bail',
+          '--debug',
+          '--kibana-install-dir', kibanaInstallDir,
+          '--',
+          '--server.maxPayloadBytes=1648576',
+        ],
+      }
+    }), {}),
+  };
+}
+
+grunt.registerTask(
+  'functionalTests:ensureAllTestsInCiGroup',
+  'Check that all of the functional tests are in a CI group',
+  async function () {
+    const done = this.async();
+
+    try {
+      const stats = JSON.parse(await execa.stderr(process.execPath, [
+        'scripts/functional_test_runner',
+        ...TEST_TAGS.map(tag => `--include-tag=${tag}`),
+        '--config', 'test/functional/config.js',
+        '--test-stats'
+      ]));
+
+      if (stats.excludedTests > 0) {
```

Comment:
Anyway to give the user information on the tests which are excluded?

Reply:
`@kbn/test` now reports excluded tests like this:
```
      14 tests in the /Users/spalger/kbn-dev/master/kibana/x-pack/test/functional/config.js config
      are excluded when filtering by the tags run on CI. Make sure that all suites are
      tagged with one of the following tags, or extend the list of tags in test/scripts/jenkins_xpack.sh

      tags: {"include":["ciGroup1","ciGroup2","ciGroup3","ciGroup4","ciGroup5","ciGroup6"],"exclude":[]}

      - dashboard mode Dashboard View Mode Dashboard viewer shows only the dashboard app link
      - dashboard mode Dashboard View Mode Dashboard viewer shows the dashboard landing page by default
      - dashboard mode Dashboard View Mode Dashboard viewer does not show the create dashboard button
      - dashboard mode Dashboard View Mode Dashboard viewer opens a dashboard up
      - dashboard mode Dashboard View Mode Dashboard viewer can filter on a visualization
      - dashboard mode Dashboard View Mode Dashboard viewer does not show the edit menu item
      - dashboard mode Dashboard View Mode Dashboard viewer does not show the view menu item
      - dashboard mode Dashboard View Mode Dashboard viewer does not show the reporting menu item
      - dashboard mode Dashboard View Mode Dashboard viewer does not show the sharing menu item
      - dashboard mode Dashboard View Mode Dashboard viewer does not show the visualization edit icon
      - dashboard mode Dashboard View Mode Dashboard viewer does not show the visualization delete icon
      - dashboard mode Dashboard View Mode Dashboard viewer shows the timepicker
      - dashboard mode Dashboard View Mode Dashboard viewer is loaded for a user who is assigned a non-dashboard mode role
      - dashboard mode Dashboard View Mode Dashboard viewer is not loaded for a user who is assigned a superuser role

```

And the output from the grunt task now looks like:

```
~/kbn-dev/master/kibana (implement/ci-groups)* $ npx grunt functionalTests:ensureAllTestsInCiGroup
Running "functionalTests:ensureAllTestsInCiGroup" task
(node:61174) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
Fatal error:
          3 tests are excluded by the ciGroup tags, make sure that
          all test suites have a "ciGroup{X}" tag and that "tasks/functional_test_groups.js"
          knows about the tag that you are using.

          tags: {"include":["ciGroup1","ciGroup2","ciGroup3","ciGroup4","ciGroup5","ciGroup6","ciGroup7","ciGroup8","ciGroup9","ciGroup10","ciGroup11","ciGroup12"]}

          - console app console app should show the default request
          - console app console app default request response should include `"timed_out" : false`
          - console app console app settings should allow changing the text size
```